pkg/operator/latencyprofilecontroller/controller.go

```diff
@@ -0,0 +1,207 @@
+package latencyprofilecontroller
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+
+	apiconfigv1 "github.com/openshift/api/config/v1"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	configv1 "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	configv1informers "github.com/openshift/client-go/config/informers/externalversions/config/v1"
+	listerv1 "github.com/openshift/client-go/config/listers/config/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	controlplanev1 "github.com/openshift/api/kubecontrolplane/v1"
+	"github.com/openshift/cluster-kube-controller-manager-operator/pkg/operator/operatorclient"
+)
+
+const (
+	controllerManagerConfigMapName = "config"
+	controllerManagerConfigMapKey  = "config.yaml"
+
+	nodeMonitorGracePeriodArgument = "node-monitor-grace-period"
+)
+
+// LatencyProfileController periodically lists the config/v1/node object
+// and fetches the worker latency profile applied on the cluster which is used to
+// updates the status of the config node object. The status updates reflect the
+// state of kube-controller-manager(s) running on control plane node(s) and their
+// observed config for node-monitor-grace-period match the applied arguments.
+type LatencyProfileController struct {
+	operatorClient  v1helpers.StaticPodOperatorClient
+	configClient    configv1.ConfigV1Interface
+	configMapClient corev1client.ConfigMapsGetter
+	nodeLister      listerv1.NodeLister
+}
+
+func NewLatencyProfileController(
+	operatorClient v1helpers.StaticPodOperatorClient,
+	configClient configv1.ConfigV1Interface,
+	nodeInformer configv1informers.NodeInformer,
+	kubeInformersForNamespaces v1helpers.KubeInformersForNamespaces,
+	kubeClient kubernetes.Interface,
+	eventRecorder events.Recorder,
+) factory.Controller {
+
+	ret := &LatencyProfileController{
+		operatorClient:  operatorClient,
+		configClient:    configClient,
+		configMapClient: v1helpers.CachedConfigMapGetter(kubeClient.CoreV1(), kubeInformersForNamespaces),
+		nodeLister:      nodeInformer.Lister(),
+	}
+
+	return factory.New().WithInformers(
+		// this is for our general configuration input and our status output in case another actor changes it
+		operatorClient.Informer(),
+
+		// We use nodeInformer for observing current worker latency profile
+		nodeInformer.Informer(),
+
+		// for configmaps of operator client target namespace
+		kubeInformersForNamespaces.InformersFor(operatorclient.TargetNamespace).Core().V1().ConfigMaps().Informer(),
+	).ResyncEvery(time.Minute).WithSync(ret.sync).ToController(
```

Comment:
I could imagine this to be a bit higher, since the observed value will be static for most of the time. But we can keep 1m  if you think it is needed.

Reply:
Initially, I had some trouble getting status updates (not facing anything similar rn); but the re-sync should IMO depend upon how long the revision roll-outs on the nodes take assuming most use cases would be 3/5 control plane nodes.

Reply:
Kept it at 5 min for the resync interval! :)